dask/array/core.py

```diff
@@ -2842,12 +2842,10 @@ def where(condition, x=None, y=None):
     x = broadcast_to(x, shape).astype(dtype)
     y = broadcast_to(y, shape).astype(dtype)
 
-    if isinstance(condition, (bool, np.bool8)):
-        if condition:
-            return x
-        else:
-            return y
+    if np.isscalar(condition):
+        return x if condition else y
     else:
+        condition = asarray(condition).astype('bool')
         return choose(condition, [y, x])
```

Comment:
I'll do a bit more targeted benchmarking and possibly open a PR to make this change. My workaround for the bug was my own implementation that wraps `np.where`, and I seem to recall it went a lot faster, but I don't have the numbers handy.